consensus/src/block_storage/block_tree.rs

```diff
@@ -373,6 +504,54 @@ impl BlockTree {
         self.commit_root_id = root_id;
     }
 
+    pub(super) fn update_window_root(&mut self, root_id: HashValue) {
+        assert!(
+            self.block_exists(&root_id),
+            "Block {} not found, previous window_root: {}",
+            root_id,
+            self.window_root_id
+        );
+        self.window_root_id = root_id;
+    }
+
+    /// `window_root` is the parent_id of the first block in the [OrderedBlockWindow](OrderedBlockWindow)
+    ///
+    /// ```text
+    ///              ┌──────────────────┐
+    ///  Genesis ──> │ A1 ──> A2 ──> A3 │ ──> A4
+    ///              └──────────────────┘
+    ///      ↑                 ↑               ↑
+    /// window_root      block_window   block_to_commit
```

Comment:
why the root is the parent of the first block instead of the first block itself?

Reply:
Fixed